### PR TITLE
Makes sure selection column stays on the left

### PR DESCRIPTION
**Pull request by alump**
Not sure if this should be merged (will talk with customer tomorrow). But makes selection column frozen, and makes sure it goes to the left when toggling selectable (now if any column have been frozen before toggling selectable, it will not go to the left).

Now for just review if code looks good, do not merge yet.

### CODE REVIEW

File: px-data-grid.html [hunk 1]
```diff
@@ -487,6 +485,15 @@
           }
         }
 
+        _handleColumnFroze(event) {
+          const columnsRow = this._vaadinGrid._columnTree[this._vaadinGrid._columnTree.length - 1].slice(0);
+          const dataColumns = columnsRow.filter(c => c.isDataColumn);
+          if (dataColumns.length) {
+            this._vaadinGrid.insertBefore(event.detail.column, dataColumns[0]);
+            this._vaadinGrid._observer.flush();
```
Comment (Toshegg) [inline px-data-grid.html]: We can remove that.

File: px-data-grid.html [hunk 2]
```diff
@@ -828,6 +835,32 @@
           }
         }
 
+        /**
+         * Make sure
```
Comment (Toshegg) [inline px-data-grid.html]: Poor method description

File: px-data-grid.html [hunk 3]
```diff
@@ -828,6 +835,32 @@
...
+            const firstRight = rightColumns[0];
+            leftColumns.forEach(c => {
+              this._vaadinGrid.insertBefore(c, firstRight);
+            });
+          }
+        }
+
+        _selectableChanged(selectable) {
+          if (selectable) {
+            setTimeout(() => {
```
Comment (Toshegg) [inline px-data-grid.html]: Can we flush vaadin-grid instead of setTimeout?

File: bower.json [hunk 4]
```diff
@@ -19,6 +19,7 @@
     "px-icon-set": "^2.0.0",
     "px-dropdown": "custom-trigger-slot",
     "vaadin-grid": "^4.1.0",
+    "vaadin-radio-button": "^v1.0.0-alpha8",
```
Comment (limonte) [inline bower.json]: - no need in `v` prefix
- no need in `-alpha8` suffix

"^1.0.0" will be enough

File: px-data-grid-selection-column.html [hunk 5]
```diff
@@ -12,8 +13,14 @@
       </template>
     </template>
     <template id="defaultBodyTemplate">
-      <vaadin-checkbox aria-label="Select Row" aria-label="Select Row" checked="{{selected}}">
-      </vaadin-checkbox>
+      <template is="dom-if" if="[[multiSelect]]">
+        <vaadin-checkbox aria-label="Select Row" aria-label="Select Row" checked="{{selected}}">
```
Comment (limonte) [inline px-data-grid-selection-column.html]: doubled `aria-label="Select Row"`

File: px-data-grid-selection-column.html [hunk 6]
```diff
@@ -12,8 +13,14 @@
...
+        </vaadin-checkbox>
+      </template>
+      <template is="dom-if" if="[[!multiSelect]]">
+          <vaadin-radio-button aria-label="Select Row" aria-label="Select Row" checked="{{selected}}">
```
Comment (limonte) [inline px-data-grid-selection-column.html]: nit: indentation

File: px-data-grid-selection-column.html (hunk 5 above)
Comment (alump) [reply px-data-grid-selection-column.html]: not doubled, just to negative dom-ifs

File: px-data-grid.html (hunk 1 above)
Comment (alump) [reply px-data-grid.html]: Why? what? sorry do not understand this.

Comment (alump) [reply px-data-grid.html]: the observer flush... o yeah, will do that

File: px-data-grid-selection-column.html (hunk 5 above)
Comment (alump) [reply px-data-grid-selection-column.html]: ok, i blind today, fixing

File: px-data-grid.html (hunk 3 above)
Comment (alump) [reply px-data-grid.html]: What would that be? Only flush I found is on test cases.

**Pull request merged**